src/components/tiles/table/use-content-change-handlers.ts

```diff
@@ -124,6 +124,10 @@ export const useContentChangeHandlers = ({
     if (!readOnly && consumerTile) {
       const sharedModelManager = consumerTile.tileEnv?.sharedModelManager;
       if (sharedModelManager?.isReady) {
+        const existingSharedTable = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, tileInfo.id);
+        if (existingSharedTable) {
+          sharedModelManager?.removeTileSharedModel(consumerTile, existingSharedTable);
+        }
```

Comment:
This deserves a comment because it's not entirely clear what the intent is. I think this is saying that if the tile we're linking to (which we're assuming is to be a consumer of this tile's DataSet) is already a provider for another DataSet, then we should break the connection between that tile and its existing DataSet. Is that the intent?